### ISSUE TITLE
Enum tag generation in definition object

### ISSUE THREAD

**Issue by samxye** [swagger-api/swagger-codegen#982]
Mustache tags for enums are not being parsed or generated. This seems to be the case only when enum is a property directly under a definition object. Here is an example from petstore:

#### - Enum tags generated

 **swagger.json**
```
  "definitions": {
    "Order": {
      "type": "object",
      "properties": {
        "status": {
          "type": "string",
          "description": "Order Status",
          "enum": [
            "placed",
            "approved",
            "delivered"
          ]
        },
      },
      "xml": {
        "name": "Order"
      }
    },
```
**corresponding "model info" output**
```
    }, {
      "baseName" : "status",
      "complexType" : "string",
      "getter" : "getStatus",
      "setter" : "setStatus",
      "description" : "Order Status",
      "datatype" : "String",
      "datatypeWithEnum" : "StatusEnum",
      "name" : "status",
      "defaultValue" : "null",
      "baseType" : "string",
      "jsonSchema" : "{\n  \"type\" : \"string\",\n  \"description\" : \"Order Status\",\n  \"enum\" : [ \"placed\", \"approved\", \"delivered\" ]\n}",
      "hasMore" : true,
      "isNotContainer" : true,
      "isEnum" : true,
      "_enum" : [ "placed", "approved", "delivered" ],
      "allowableValues" : {
        "values" : [ "placed", "approved", "delivered" ]
      }
    }, {
```

#### - Enum tags not generated

**swagger.json**
```
  "definitions": {
    "status": {
      "type": "string",
      "description": "Order Status",
      "enum": [
         "placed",
         "approved",
         "delivered"
      ]
    },
```
**corresponding "model info" output**
```
[ {
  "model" : {
    "name" : "StatusType",
    "classname" : "StatusType",
    "description" : "Order Status",
    "classVarName" : "StatusType",
    "modelJson" : "{\n  \"type\" : \"string\",\n  \"description\" : \"Order Status\"\n}",
    "vars" : [ ],
    "imports" : [ ],
    "emptyVars" : true
  },
  "importPath" : "StatusType"
},
```

**Comment by samxye** [comment]
Are there plans on implementing this anytime soon? If I can get a rough outline of an approach to take, I can submit a pull request for an implementation

**Comment by wing328** [comment]
@samxye sorry for late reply on this. Do you mind testing with the latest swagger-codegen to see if the issue persists?

**Comment by samxye** [comment]
@wing328 still not working unfortunately.

**Comment by wing328** [comment]
@samxye for the following
```
  "definitions": {
    "status": {
      "type": "string",
      "description": "Order Status",
      "enum": [
         "placed",
         "approved",
         "delivered"
      ]
    },
```
Is that the definition of a model? It seems to be missing the `properties`

**Comment by fehguy** [comment]
This is addressed by #1740
